tests/conftest.py

```diff
@@ -364,9 +364,12 @@ def nbrhosts(ansible_adhoc, tbinfo, creds, request):
     Shortcut fixture for getting VM host
     """
 
+    devices = {}
+    if not tbinfo['vm_base']:
```

Comment:
Should there also be an AND check for `topo=="tgen"`?
If it so happens that `vm_base` is missing for non `tgen` topos then something else is wrong, and it should be gracefully raised as an error?

Reply:
Also, a log `INFO` line here indicating that vm_base is empty can help in debugging for non tgen topos.

Reply:
Done!